pkg/azureutils/azure_wrap.go

```diff
@@ -2,284 +2,8 @@ package azureutils
 
 import (
 	"context"
-	"fmt"
-	"net/http"
-	"strings"
-	"sync"
-	"time"
-
-	armnetwork "github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
-
-	"k8s.io/klog/v2"
-	"k8s.io/utils/pointer"
-)
-
-const (
-	// ID string used to create a not existing PLS placehold in plsCache to avoid redundant
-	PrivateLinkServiceNotExistID = "PrivateLinkServiceNotExistID"
 )
 
-var (
-	vmCacheTTLDefaultInSeconds           = 60
-	loadBalancerCacheTTLDefaultInSeconds = 120
-	nsgCacheTTLDefaultInSeconds          = 120
-	routeTableCacheTTLDefaultInSeconds   = 120
-	publicIPCacheTTLDefaultInSeconds     = 120
-	plsCacheTTLDefaultInSeconds          = 120
-)
-
-func (az *Cloud) newVMCache() (*TimedCache, error) {
-	getter := func(key string) (interface{}, error) {
-		// Currently InstanceView request are used by azure_zones, while the calls come after non-InstanceView
-		// request. If we first send an InstanceView request and then a non InstanceView request, the second
-		// request will still hit throttling. This is what happens now for cloud controller manager: In this
-		// case we do get instance view every time to fulfill the azure_zones requirement without hitting
-		// throttling.
-		// Consider adding separate parameter for controlling 'InstanceView' once node update issue #56276 is fixed
-		ctx, cancel := getContextWithCancel()
-		defer cancel()
-
-		resourceGroup, err := az.GetNodeResourceGroup(key)
-		if err != nil {
-			return nil, err
-		}
-
-		res, verr := az.VMClient.Get(ctx, resourceGroup, key, nil)
-		vm := res.VirtualMachine
-		exists, rerr := checkResourceExistsFromError(&Error{
-			RawError: verr,
-		})
-		if rerr != nil {
-			return nil, rerr.Error()
-		}
-
-		if !exists {
-			klog.V(2).Infof("Virtual machine %q not found", key)
-			return nil, nil
-		}
-
-		if vm.Properties != nil &&
-			strings.EqualFold(pointer.StringDeref(vm.Properties.ProvisioningState, ""), string(ProvisioningStateDeleting)) {
-			klog.V(2).Infof("Virtual machine %q is under deleting", key)
-			return nil, nil
-		}
-
-		return &vm, nil
-	}
-
-	if az.VMCacheTTLInSeconds == 0 {
-		az.VMCacheTTLInSeconds = vmCacheTTLDefaultInSeconds
-	}
-	return NewTimedcache(time.Duration(az.VMCacheTTLInSeconds)*time.Second, getter)
-}
-
-// checkExistsFromError inspects an error and returns a true if err is nil,
-// false if error is an autorest.Error with StatusCode=404 and will return the
-// error back if error is another status code or another type of error.
-func checkResourceExistsFromError(err *Error) (bool, *Error) {
-	if err == nil {
-		return true, nil
-	}
-
-	if err.HTTPStatusCode == http.StatusNotFound {
-		return false, nil
-	}
-
-	return false, err
-}
-
 func getContextWithCancel() (context.Context, context.CancelFunc) {
```

Comment:
Do we use this function? If so, lets move this to some other utils file and remove this file as well.